flink-staging/flink-ml/src/main/scala/org/apache/flink/ml/package.scala

```diff
@@ -70,6 +70,15 @@ package object ml {
       dataSet.map(new BroadcastSingleElementMapperWithIteration[T, B, O](dataSet.clean(fun)))
         .withBroadcastSet(broadcastVariable, "broadcastVariable")
     }
+
+    /** Calculates the mean value of a DataSet[T <\: Numeric[T]]
+      *
+      * @return A DataSet[Double] with the mean value as its only element
+      */
+    def mean()(implicit num: Numeric[T], ttit: TypeInformation[(T, Int)]): DataSet[Double] =
```

Comment:
I think this method should be placed in a `NumericRichDataSet[T: Numeric]` class. Otherwise we provide a `mean` method for `DataSets` whose element type does not support this operation. This might be misleading IMO.